scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py

```diff
@@ -0,0 +1,521 @@
+        Default is 1 (recommended in [1]_, p 19).
+    initial_barrier_parameter: float, optional
+        Initial barrier parameter. Used only when inequality constraints
+        are present. Default is 0.1 (recommended in [1]_ p. 19).
+    initial_tolerance: float, optional
+        Initial subproblem tolerance. Used only when inequality constraints
+        are present. Default is 0.1 (recommended in [1]_ p. 19).
+    factorization_method : string or None, optional
+        Method to factorize the Jacobian of the constraints.
+        Use None (default) for the auto selection or one of:
+
+            - 'NormalEquation'
+            - 'AugmentedSystem'
+            - 'QRFactorization'
+            - 'SVDFactorization'
+
+        The factorization methods 'NormalEquation' and 'AugmentedSystem'
+        should be used only when ``sparse_jacobian=True``. The projections
+        required by the algorithm will be computed using, respectively,
+        the the normal equation and the augmented system approach explained
+        in [1]_. 'NormalEquation' computes the Cholesky factorization of
+        ``(A A.T)`` and 'AugmentedSystem' performs the LU factorization
+        of an augmented system. They usually provide similar results.
+        'NormalEquation' requires scikit-sparse installed. 'AugmentedSystem'
+        is used by default for sparse matrices. The methods 'QRFactorization'
+        and 'SVDFactorization' should be used when ``sparse_jacobian=False``.
+        They compute the required projections using, respectively,
+        QR and SVD factorizations. The 'SVDFactorization' method can cope
+        with Jacobian matrices with deficient row rank and will be used
+        whenever other factorization methods fail (which may imply the
+        conversion of sparse matrices to a dense format when required).
+        By default uses 'QRFactorization' for  dense matrices.
+    finite_diff_rel_step : None or array_like, optional
+        Relative step size for the finite difference approximation.
+    maxiter : int, optional
+        Maximum number of algorithm iterations. Default is 1000.
+    verbose : {0, 1, 2}, optional
+        Level of algorithm's verbosity:
+
+            * 0 (default) : work silently.
+            * 1 : display a termination report.
+            * 2 : display progress during iterations.
+            * 3 : display progress during iterations (more complete report).
+
+    disp : bool, optional
+        If True (default) then `verbose` will be set to 1 if it was 0.
+
+    Returns
+    -------
+    `OptimizeResult` with the following fields defined:
+    x : ndarray, shape (n,)
+        Solution found.
+    fun : float
+        Objective function at the solution.
+    grad : ndarray, shape (n,)
+        Gradient of the objective function at the solution.
+    lagrangian_grad : ndarray, shape (n,)
+        Gradient of the Lagrangian function at the solution.
+    nfev : integer
```

Comment:
Judging from past experience the user would like to know the total number of function evaluations, including those used in numerical calculation of grad and hess.

Reply:
A few commits ago I was using conceptual nfev/ngev/nhess... I changed it because I think  it make more sense to give the real number of function/gradient/hessian evaluations. It does have a clear meaning, it is easier to document, and I think it is more useful for the user.

Besides that, I think the number of iterations ``niter``, gives some information about the algorithm progress, and makes "conceptual function evaluations" more or less redundant.